---
title: "Gene enrichment"
author: "Simon Urbanek"
date: "2023-10-13"
output: pdf_document
categories: quantitative biology
tags:
- transcriptomics
- genes
slug: "gene-enrichment"
---

# Enrichment analysis

Having obtained a file with the results of a DSeq2 differential expression calculation (stored below in the variable `datafile`), we perform an enrichment analysis in different databases.

Check extra info ![here](https://rpubs.com/Alexis22/ClusterProfiler)

## Install packages

```{r}

# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("clusterProfiler")
# BiocManager::install("org.Hs.eg.db")
# BiocManager::install("ReactomePA")
# 
# install.packages("enrichplot", repos = "https://cran.r-project.org")
# install.packages("enrichR", repos = "https://cran.r-project.org")
# install.packages("crayon", repos = "https://cran.r-project.org")
# install.packages("GOSemSim")
# install.packages("DOSE")

```
## Load libraries
```{r}

library(clusterProfiler)
library(org.Hs.eg.db)
# library(GOSemSim)
# library(DOSE)
# library(enrichR)
# library(crayon) # So messages can be in blue font
library(enrichplot)
```
## Load the list of genes
```{r}
datafile <- 'data/Control_vs_NPDR.tsv'
df <- read.csv(datafile,header=TRUE,sep='\t')
head(df)
# obtain the log2foldchage column
fold_change <- df$log2FoldChange
# name the vector
names(fold_change) <- df$ext_gene

genes <- mapIds(org.Hs.eg.db, df$ext_gene, 'ENTREZID', 'SYMBOL', multiVals = "first")
names(fold_change) <- genes
head(fold_change)
# sort the list in dicreasing order, required for clusterprolfier
gene_list <- sort(fold_change, decreasing = TRUE)
head(gene_list)
output_dir <- 'results'
```


## Run the analysis on different Ontologies

### Perform analysis on enrichKEGG
```{r}
KEGG_genes <- enrichKEGG(gene= genes, organism= "hsa", pvalueCutoff = 0.05)
head(KEGG_genes)
dotplot(KEGG_genes,showCategory=10)
sm_kegg <- pairwise_termsim(KEGG_genes)
enrichplot::emapplot(sm_kegg, showCategory = 50)
```

```{r}
# ck_kegg <- compareCluster(gene_list, fun="enrichKEGG", pvalueCutoff=0.05)
# sm_kegg <- pairwise_termsim(ck_kegg)
# dotplot(ck_kegg, size='count', showCategory=10, title='enrichKEGG')
# emapplot(sm_kegg)
```

### Perfomr GSEA

```{r}
GSEA_genes <- gseKEGG(geneList     = gene_list, organism     = "hsa", nPerm= 1000,         minGSSize    = 10, pvalueCutoff = 1, verbose      = FALSE)
head(GSEA_genes)
gseaplot(GSEA_genes,geneSetID=1,title=GSEA_genes$Description[1])
gseaplot(GSEA_genes,geneSetID=2,title=GSEA_genes$Description[2])
gseaplot(GSEA_genes,geneSetID=3,title=GSEA_genes$Description[3])
gseaplot(GSEA_genes,geneSetID=4,title=GSEA_genes$Description[4])
```

### Perform analysis on enrichPathway


```{r}
ck_pathway <- compareCluster(gene_list, fun="enrichPathway", pvalueCutoff=0.05)
ck_pathway
sm_pathway <- pairwise_termsim(ck_pathway)
dotplot(ck_pathway, size='count', showCategory=10, title='enrichPathway')
sm_pathway
emapplot(sm_pathway)
storage_pathway <- paste(output_dir, 'clusterProfiler_enrichments_RP.csv', sep='/')
write.csv(ck_pathway@compareClusterResult, storage_pathway)
```

### Perform analysis on enrichGO
```{r}
ck_go <- compareCluster(gene_list, fun="enrichGO", pvalueCutoff=0.05, OrgDb=org.Hs.eg.db)
sm_go <- pairwise_termsim(ck_go)
dotplot(ck_go, size='count', showCategory=10, title='enrichGO')
emapplot(sm_go)
```


### Perform analysis on enrichDO
```{r}
ck_do <- compareCluster(gene_list, fun="enrichDO", pvalueCutoff=0.05)
sm_do <- pairwise_termsim(ck_do)
dotplot(ck_do, size='count', showCategory=5, title='enrichDO')
emapplot(sm_do)
```