---
title: "Gene enrichment"
author: "Jordi Vill√†-Freixa"
date: "2023-10-13"
categories: ["R"]
tags: ["transcriptomics", "gene-enrichment","R Markdown"]
---

# Enrichment analysis

Having obtained a file with the results of a DSeq2 differential expression calculation (stored below in the variable `datafile`), we perform an enrichment analysis in different databases.

Check extra info ![here](https://rpubs.com/Alexis22/ClusterProfiler)

## Install packages

```{r}

# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("clusterProfiler")
# BiocManager::install("org.Hs.eg.db")
# 
# install.packages("enrichplot", repos = "https://cran.r-project.org")


```

## Load libraries

```{r}
library(clusterProfiler)
library(org.Hs.eg.db)
library(enrichplot)
```

## Load the list of genes

```{r}
datafile <- 'data/DESeq2_FC.tsv'
df <- read.csv(datafile,header=TRUE,sep='\t')
# obtain the log2foldchage column
fold_change <- df$log2FoldChange
# name the vector
genes <- mapIds(org.Hs.eg.db, df$ext_gene, 'ENTREZID', 'SYMBOL', multiVals = "first")
names(fold_change) <- genes
head(fold_change)
# sort the list in dicreasing order, required for clusterprolfier
gene_list <- sort(fold_change, decreasing = TRUE)
head(gene_list)
output_dir <- 'results'
```

## Run the analysis on different Ontologies

### Perform analysis on enrichKEGG

```{r}
KEGG_genes <- enrichKEGG(gene= genes, organism= "hsa", pvalueCutoff = 0.05)
head(KEGG_genes)
dotplot(KEGG_genes,showCategory=15)
sm_kegg <- pairwise_termsim(KEGG_genes)
enrichplot::emapplot(sm_kegg, showCategory = 20)
```

### Perform GSEA

```{r}
GSEA_genes <- gseKEGG(geneList     = gene_list, organism     = "hsa", nPerm= 1000,         minGSSize    = 10, pvalueCutoff = 1, verbose      = FALSE)
head(GSEA_genes)
gseaplot(GSEA_genes,geneSetID=1,title=GSEA_genes$Description[1])
gseaplot(GSEA_genes,geneSetID=2,title=GSEA_genes$Description[2])
gseaplot(GSEA_genes,geneSetID=3,title=GSEA_genes$Description[3])
gseaplot(GSEA_genes,geneSetID=4,title=GSEA_genes$Description[4])
```

### Perform analysis on enrichGO

```{r}
head(genes)
ck_go <- enrichGO(genes, 'org.Hs.eg.db', keyType = "ENTREZID",ont="All", pvalueCutoff=0.01)
head(ck_go)
dotplot(ck_go, showCategory=15, title='enrichGO')
```
